### ISSUE TITLE
Append "Shape" suffix to SHACL shape in shaclgen.py

### ISSUE THREAD

**Issue by robertschubert** [linkml/linkml#2037]
**Is your feature request related to a problem? Please describe.**
We use linkml to genereate OWL classes as well as SHACL shapes. Those are then written to a graph db.
And this is where our problem arises:
There is a name clash between the OWL class and the SHACL shape since they have the same name (according to the name specified in the linkml yaml file).

**Describe the solution you'd like**
My suggestion would be to append the keyword "Shape" to a SHACL shape. 
Concrete example:
```
...
classes:
  MyClass:
    title: 'title of my class'
...

```

Currently following SHACL shape is generated:
```
prefix:MyClass a sh:NodeShape ;
    sh:closed false ;
...
    sh:targetClass prefix:MyClass .
```

Suggestion for the future:
Append "Shape" to MyClass and point to targetClass MyClass (without "Shape"):
```
prefix:MyClassShape a sh:NodeShape ;
    sh:closed false ;
...
    sh:targetClass prefix:MyClass .
```

**How important is this feature?** Select from the options below:
• Medium - can do work without it; but it's important (e.g. to save time or for convenience)
--> we are able to patch the shaclgen.py with the proposed code in the PR.

**When will use cases depending on this become relevant?** Select from the options below:
• Short-term - 2-4 weeks

**Additional context**
I have already implemented a solution and will open a pull request for this feature.

**Comment by cmungall** [comment]
Thanks! Having the option to append `Shape` will better adhere to shacl conventions.

Is there a good quick guide that explains the rationale for having separate URIs, and what kinds of problems people might expect in punning a shape and a class? IMO the way most people use semweb vocabs like schema.org, prov, etc "classes" are more like OO-classes/frames and less like OWL classes....

**Comment by robertschubert** [comment]
> Thanks! Having the option to append `Shape` will better adhere to shacl conventions.
> 
> Is there a good quick guide that explains the rationale for having separate URIs, and what kinds of problems people might expect in punning a shape and a class? IMO the way most people use semweb vocabs like schema.org, prov, etc "classes" are more like OO-classes/frames and less like OWL classes....

No problem :) I agree that I mostly see a "Shape" suffix for SHACL shapes so this feature will really be useful.
I have not a good reference to a quick guide. In our case we use an owl class for the ontology definition and link the SHACL shape via targetClass to that owl class. The SHACL is then of cource used for schema validation.

**Comment by robertschubert** [comment]
set the linked PR to "ready for review"

**Comment by cmungall** [comment]
Fixed by #2038
